### PR TITLE
Create form for managing Facebook data for the preview

### LINKED ISSUE #443: Create form for managing the Facebook (preview) data
For managing the Facebook data and reflecting this data in the preview we need a form where we can enter:
- The title
- The description
- The image

This form should manage the values in the #440 

### How it looks like
Screen Shot 2020-02-04 at 10 01 30

### The elements
- [x] Create an input element for the title. 
   - [x] The label has `Facebook title` as value
   - [x] This element should be support inserting variables (these are the replace_vars in the code). 
   - [x] When the value has changed it should trigger the action for updating the title in the store.
- [x] Create an textarea for the description. 
   - [x] The label has `Facebook description` as value
   - [x] Just like the title element this element should have support for inserting variables as well. 
   - [x] When the value of the textarea is changed it should trigger the action for updating the description in the store.
- [x] Create a button for setting the image. This element can use the media library in WordPress for selecting an image. 
   - [x] The label has `Facebook image` as value
   - [x] When an image is selected it should trigger the action for updating the image in the store. 

Parent https://github.com/Yoast/javascript/issues/444

### PR DESCRIPTION
**Pull request by bintzandt**
## Summary
This PR adds the following components:
* `SocialPreviewFormWrapper.js` -> A wrapper to display the form in the components app.
* `ImageUpload.js` -> A generic component that displays a Title, an optional Warning Alert and a button that can be used to upload / select a file. Note that this component is unaware of its context. WP specefic functionality should be provided via a wrapper (i.e. the `SocialMetadataPreviewForm.js`).
* `SocialMetadataPreviewForm.js` -> Barebones component that displays a form with an image upload and a draft-js input field for the title and description.

changelog: N/A

## Relevant technical choices:
* `ImageUpload.js` has been kept as minimal as possible so that it can be re-used in other places. 
* Styling is done in a [separate issue](https://github.com/Yoast/javascript/issues/486).
* No tests have been written for the `SocialMetadataPreviewForm.js` because I was unable to get it rendered by the `react-test-renderer`.
* I wanted to add a test that simulates a button click using `enzyme` but I was unable to get `shallow` to render anything. Attempts using `ReactTestUtils` were also unsuccessful.
* There is no specification of the Replacement variables that should be added to the component. This is not really a problem since they can easily be added when we are making the actual implementation.

## TODO
- [x] Provide the correct defaults as placeholder.
- [x] Make sure that all the methods / callbacks can be provided separately from the context (WordPress or something else). 
- [x] Complete this PR with all the details.

## Test instructions (TODO)
This PR can be tested by following these steps:
* The best way to test this PR is by having a look at the Components app. I have added a new page called `Social Previews data form` which shows the current state of the forms components.

## Quality assurance

* [x] I have tested this code to the best of my abilities
* [x] I have added unittests to verify the code works as intended (see the note above about the `SocialMetadataPreviewForm`).

Fixes #443

### CODE REVIEW

File: apps/components/SocialPreviewFormWrapper.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+const recommendedReplacementVariables = [
+	"title",
+	"post_type",
+];
+
+const selectFileClick = () => {
+	alert( "YOU CLICKED MY BUTTON!" );
+};
+
+const SocialPreviewFormWrapper = () =>
```
Comment (maartenleenders) [inline apps/components/SocialPreviewFormWrapper.js]: Seems to me that this is the `FacebookFormWrapper`, or `FacebookForm`, correct?

File: packages/social-metadata-previews/src/shared/ImageUpload.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * @param {object} props The properties passed to this component.
+ * @param {string} props.title The title that is displayed above the selection button.
+ * @param {string[]} props.warnings An array of warnings that detail why the image cannot be used.
+ * @param {function} props.onClick Function that specifies what happens when the button is clicked.
+ *
+ * @returns {React.Component} A fragment with a title, optional warnings and an image selection button.
+ */
+const ImageSelect = ( { title, warnings, onClick } ) =>
+	<Fragment>
+		<p>
```
[the remark this answers is not in the document]
Comment (bintzandt) [reply packages/social-metadata-previews/src/shared/ImageUpload.js]: This will be done in https://github.com/Yoast/javascript/issues/487.

**Comment by maartenleenders** [comment]
Acceptance 🚧 

I am missing some functionality the `SnippetEditorFields` do have:
- [ ] Clicking label will focus input field.
- [ ] The inputfield is only one line high, but the description div is higher. In the snippeteditor, clicking in the bottom of the input wrapper div (so outside the input) will still focus the input. Here, that does not work.
- [ ] Focused fields will have a blue outline.

I am well aware that this could be done through state in the final wrapper component, that combines the preview and the editor. Because both components need information on whichever field is active, and both need control of the focus.

However, now the `SocialMetadataPreviewForm` is not self-dependent. I would like it to have some control over its focus, to be more standalone. But perhaps that is too much for this PR, or even project?

**Comment by Xyfi** [comment]
About the multiple warnings: show multiple alert components

**Comment by maartenleenders** [comment]
The focussing of individual items in the form (state management) will be done in a _to be created_ issue to create the parent container for the form and the preview.

**Pull request merged**